### DISCUSSION TITLE
How to integrate with outer database

### DISCUSSION THREAD

**Discussion by hidech** [webstudio-is/webstudio-community#28] [Q&A] [answered]
Hi, I'm in Japan and I have encounted Webstudio, and I could also make this page quickly. https://webstudiosample.wstd.io/
This is amazing!

I have build several web application with Bubble or Wordpress. These products are good enough, but I want to design free I wanted, and they are not suitable. Now I'm creating apps with FlutterFlow and Supabase, but it is not compatible with SEO because of non-writing html tags.

My question is, is Webstudio preparing or having ability to integrate outer database? I want to create Blog or some kind of social media. If possible, how to do it? I found "code" element in Webstudio, but I cannot find instruction of it. I can handle JavaScript or query language.

Thanks in advance, I'm really looking forward to the future of Webstudio!

**Comment by kof** [comment]
Absolutely, integrations with any data source: API, database etc is the core of the architecture.

**Reply by kof** [reply]
Code component is for code snippets, so you can format code

**Reply by hidech** [reply]
Thanks! I got it.
Is there a way to execute JavaScript code for now?

**Reply by kof** [reply, marked as answer]
We will add Embed component soon, I understand this is critical, pls follow this issue https://github.com/webstudio-is/webstudio-builder/issues/1563

**Reply by hidech** [reply]
Thanks, I know! I'll follow that topic.
